File: src/test/java/ru/org/linux/gallery/ImageDaoIntegrationTest.java
/*
 * Copyright 1998-2012 Linux.org.ru
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */
package ru.org.linux.gallery;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.ContextHierarchy;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;

import java.util.List;

import static org.junit.Assert.assertEquals;

@RunWith(SpringJUnit4ClassRunner.class)
@ContextHierarchy({
        @ContextConfiguration("classpath:database.xml"),
        @ContextConfiguration(classes = ImageDaoIntegrationTestConfiguration.class)
})
public class ImageDaoIntegrationTest {

  @Autowired
  private ImageDao imageDao;

  /**
   * Проверка galleryDao.getGalleryItems().
   * Проверяется, что метод вернёт 3 объекта GalleryDto.
   */
  @Test
  public void getGalleryItemsTest() {
    List<GalleryItem> galleryDtoList = imageDao.getGalleryItems(3);
    assertEquals(3, galleryDtoList.size());
  }
}

File: src/test/scala/ru/org/linux/gallery/ImageDaoIntegrationTest.scala
/*
 * Copyright 1998-2026 Linux.org.ru
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package ru.org.linux.gallery

import munit.FunSuite
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.{ContextConfiguration, ContextHierarchy}
import ru.org.linux.test.SpringTestSupport

import java.util.Date

@ContextHierarchy(
  Array(
    new ContextConfiguration(value = Array("classpath:database.xml")),
    new ContextConfiguration(classes = Array(classOf[ImageDaoIntegrationTestConfiguration]))
  ))
class ImageDaoIntegrationTest extends FunSuite with SpringTestSupport:

  @Autowired
  var imageDao: ImageDao = scala.compiletime.uninitialized

  test("getGalleryItemsTest"):
    val galleryDtoList = imageDao.getGalleryItems(3)
    assertEquals(galleryDtoList.size, 3)

    val commitDates = galleryDtoList.map(_.commitDate)
    val sortedDesc = commitDates.sorted(using Ordering[Date].reverse)
    assertEquals(commitDates, sortedDesc, "gallery items must be ordered by commitdate DESC")

end ImageDaoIntegrationTest
